cartoframes/data/services/isolines.py

```diff
@@ -148,11 +156,14 @@ def _iso_areas(self,
                 raise Exception('No valid geometry found. Please provide an input source with ' +
                                 'a valid geometry or specify the "geom_col" param with a geometry column.')
 
-            to_carto(source_cdf, temporary_table_name, self._credentials, log_enabled=False)
+            index_as_cartodbid = CARTO_INDEX_KEY not in source_cdf.columns
+
+            to_carto(source_cdf, temporary_table_name, self._credentials, index=index_as_cartodbid,
+                     index_col=CARTO_INDEX_KEY, log_enabled=False)
             source_query = 'SELECT * FROM {table}'.format(table=temporary_table_name)
 
-        source_columns = source_manager.get_column_names()
-        source_has_id = 'cartodb_id' in source_columns
+        if source_col is None:
+            source_col = CARTO_INDEX_KEY
```

Comment:
If this is the default value, we could set it in the method signature and avoid this conditional.

Reply:
I didn't want to add this to the header because of the documentation (reference). The index is set as cartodb_id when there is no cartodb_id column in the dataframe, but this is something I want to keep internally. Does it make sense to you?

Reply:
OK, makes sense then.